packages/lib-classifier/src/store/WorkflowStepStore/WorkflowStepStore.js

```diff
@@ -84,15 +85,17 @@ const WorkflowStepStore = types
   }))
   .actions(self => {
     function _onWorkflowChange() {
-      if (self.workflow) {
+      // Reset steps and tasks when the workflow ID or version change.
+      if (self.workflow?.id && self.workflow?.version) {
         self.reset()
         self.setStepsAndTasks()
       }
     }
 
     function _onLocaleChange() {
-      if (self.locale && self.workflow?.strings) {
-        self.setTaskStrings()
+      // Update task language strings when the workflow language changes.
```

Comment:
```suggestion
      // Update task translation strings when the classifier's locale changes.
```

Reply:
This function reacts to `self.workflow.strings`. If it reacts to the store locale, then it fires before the workflow strings change. Now that `workflow.strings` is observable, we can observe it directly for changes. 

Maybe I should rename it?